node/src/components/consensus/highway_core/highway.rs

```diff
@@ -304,6 +312,9 @@ impl<C: Context> Highway<C> {
         match vertex {
             Vertex::Vote(vote) => {
                 let v_id = self.validator_id(&vote).ok_or(VoteError::Creator)?;
+                if vote.wire_vote.instance_id != self.instance_id {
+                    return Err(VoteError::InstanceId.into());
+                }
```

Comment:
We need to validate the `Evidence` below as well: In case of an equivocation let's check that the two votes have the same sequence number, creator and instance ID (although not necessarily _our_ instance ID).

Reply:
>(although not necessarily our instance ID).

How is that? Isn't instance ID defining a "fork"? How can we be validating messages that don't belong to our fork?

Reply:
Not only fork. Could be a previous era, for example. But regardless, an equivocation is an equivocation and proves that someone in possession of a certain private key is misbehaving,

Reply:
Also, should we check that the signatures are correct as well, or is that being checked elsewhere?

Reply:
Exactly: It's meant to allow us punishing for equivocation in a previous era, but effectively it allows us to punish you if you equivocated with that key in any network — which is fine, I think?

Reply:
I don't know what's the point of punishing a validator if he/she equivocated in a different network. I don't think we should be doing that.

Reply:
It's as if Poland was punishing me for my crimes in Ecuador.

Reply:
Then we'd either need a separate network ID in `Highway`, or somehow know about the last few `unbonding_period` instance IDs.

Reply:
We already know which eras are still bounded.

Reply:
Although, If we don't do this check doesn't it mean that we will be reporting all equivocators, in all eras, in all forks etc etc as part of `SystemTransactions`?

Reply:
> We already know which eras are still bounded.

Sure, but `Highway` currently doesn't, so we'll have to add it.

> doesn't it mean that we will be reporting all equivocators, in all eras, in all forks etc etc as part of `SystemTransactions`?

Yes… but there's still a few pieces missing for that anyway: Currently `Highway` only has the current era's public keys, so it doesn't even know other validators' IDs.